src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamCertificateTrustTests.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.IO.Tests;
+using System.Security.Authentication;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    using Configuration = System.Net.Test.Common.Configuration;
+
+    public class SslStreamCertificateTrustTest
+    {
+        [Fact]
+        // not supported on Windows, not implemented elsewhere
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public async Task SslStream_SendCertificateTrust_CertificateCollection()
+        {
+            (SslStream client, SslStream server) = TestHelper.GetConnectedSslStreams();
+            using (client)
+            using (server)
+            using (X509Certificate2 serverCertificate = Configuration.Certificates.GetServerCertificate())
+            using (X509Certificate2 clientCertificate = Configuration.Certificates.GetClientCertificate())
+            {
+                SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions
+                {
+                    ServerCertificate = serverCertificate,
+                    ClientCertificateRequired = true,
+                    RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true,
+                    ServerCertificateContext = SslStreamCertificateContext.Create(
+                        serverCertificate,
+                        null,
+                        trust: SslCertificateTrust.CreateForX509Collection(
+                            new X509Certificate2Collection { serverCertificate, clientCertificate },
+                            sendTrustInHandshake: true))
+                };
+
+                string[] acceptableIssuers = Array.Empty<string>();
+                SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
+                {
+                    TargetHost = "localhost",
+                    EnabledSslProtocols = SslProtocols.Tls12,
+                    RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true,
+                    LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, issuers) =>
+                    {
+                        if (remoteCertificate == null)
+                        {
+                            // ignore the first call, we should receive acceptable issuers in the next one
+                            return null;
+                        }
+
+                        acceptableIssuers = issuers;
+                        return clientCertificate;
```

Comment:
This _probably_ does not matter but I'm wondering if we can turn this into Theory and have bool to send the client cert conditionally.

Reply:
I think this does not matter for this test, we are just checking that `issuers` are populated based on the `SslCertificateTrust`. There are other tests that exercise the cert-returning behavior